### PR TITLE
Add FixAll in Solution support for Enable nullable refactoring

### LINKED ISSUE #65852: Automating nullable enable over thousands of csproj
**Brief description:**

I have a couple of thousand csproj and millions of lines of code in our repository. I'd like to start the journey of moving to `#nullable enable` everywhere, but the current analyzers require that I insert `#nullable restore` into a file and perform operations through the VS IDE. I'm looking for a way to make large changes by invoking analyzers from the command line. I've done this in the past with dotnet-format.

To start, I'd like to invoke these two:

"Enable Nullable reference types in project"
"Remove Redundant Nullable directive"

**Languages applicable:**

C# only

@sharwell @mjsabby

### PR DESCRIPTION
**Pull request by mavasani**
This refactoring was identified as a high value refactoring for FixAll in solution support.

Fixes #65852

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.FixAllProvider.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    cancellationToken => createChangedSolution(CodeActionPurpose.Apply, cancellationToken),
+                    nameof(CSharpFeaturesResources.Enable_nullable_reference_types_in_solution))
+                {
+                    _createChangedSolution = createChangedSolution;
+                }
+
+                protected override async Task<IEnumerable<CodeActionOperation>> ComputePreviewOperationsAsync(CancellationToken cancellationToken)
+                {
+                    var changedSolution = await _createChangedSolution(CodeActionPurpose.Preview, cancellationToken).ConfigureAwait(false);
+                    if (changedSolution is null)
```
Comment (Youssef1313) [inline src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.FixAllProvider.cs:72]: Is this reachable?

Comment (mavasani) [reply src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.FixAllProvider.cs:72]: I have kept this consistent with the code action registered for the original refactoring: https://github.com/dotnet/roslyn/blob/d1427b1a828a4a4a83a2bd6b2fce6c5e3eff674c/src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs#L255-L276

**Comment by mavasani** [comment]
Ping @dotnet/roslyn-ide for reviews

**Review by sharwell — APPROVED** [review]

**Comment by mavasani** [comment]
Thank you @sharwell

**Pull request merged**